zerver/views/users.py

```diff
@@ -203,6 +213,122 @@ def reactivate_user_backend(
 )
 
 
+def filter_changes_in_profile_data(
+    recipient_user: UserProfile,
+    old_profile_data: List[ProfileDataElement],
+    new_profile_data: List[ProfileDataElement],
+) -> Tuple[List[Any], List[Any]]:
+    clean_old_profile_data = []
+    clean_new_profile_data = []
+
+    for field in new_profile_data:
+        id = field["id"]
+        index = id - 1
+        new_value = field["value"] if field["value"] else None
+        old_value = old_profile_data[index]["value"] if old_profile_data[index]["value"] else None
+
+        if (new_value is None and old_value is None) or new_value == old_value:
+            continue
+
+        # For some fields, the value doesn't provide a useful
+        # data to display. So we should catch these fields and
+        # perform extra operations to fetch a more useful data
+        # for the users.
+
+        # This is the id for the field "Favourite code editor"
+        # The value will return "0" or "1" representing the data
+        # Thus, we should use the value and find the name of the
+        # text editor.
+        if id == 4:
+            if new_value is not None:
+                new_field_data = orjson.loads(field["field_data"])
+                new_value = str(new_field_data[new_value]["text"])
+            if old_value is not None:
+                old_field_data = orjson.loads(old_profile_data[index]["field_data"])
+                old_value = str(old_field_data[old_value]["text"])
+
+        # This is the id for the field "Mentors"
+        # The value will return a list of user ids.
+        # Thus, we should use the user id and fetch for the user
+        # profile and output them as silent mentions
+        elif id == 7:
+            if new_value is not None:
+                temp_mentor_string = ""
+                for mentor_user_id in new_value:
+                    assert isinstance(mentor_user_id, int)
+                    mentor_user_profile = access_user_by_id(
+                        recipient_user,
+                        mentor_user_id,
+                        allow_deactivated=True,
+                        allow_bots=True,
+                        for_admin=False,
+                    )
+                    temp_mentor_string += silent_mention_syntax_for_user(mentor_user_profile) + " "
+                new_value = temp_mentor_string
+            if old_value is not None:
+                temp_mentor_string = ""
+                for mentor_user_id in old_value:
+                    assert isinstance(mentor_user_id, int)
+                    mentor_user_profile = access_user_by_id(
+                        recipient_user,
+                        mentor_user_id,
+                        allow_deactivated=True,
+                        allow_bots=True,
+                        for_admin=False,
+                    )
+                    temp_mentor_string = silent_mention_syntax_for_user(mentor_user_profile) + " "
+                old_value = temp_mentor_string
```

Comment:
Since these two operations are very similar, I wonder if I could convert this into a function inside the current function.